src/careamics/lightning/lightning_module.py

```diff
@@ -299,16 +315,15 @@ def training_step(self, batch: Tensor, batch_idx: Any) -> Any:
         Any
             Loss value.
         """
-        x, *aux = batch
+        x, *aux = batch  # TODO: check what is a `batch`
         out = self.model(x)
         target = aux[0]
+
+        # Update loss parameters
         # TODO rethink loss parameters
         self.loss_parameters.current_epoch = self.current_epoch
-        self.loss_parameters.inputs = x
-        self.loss_parameters.mask = ~((target == 0).reshape(len(target), -1).all(dim=1))
-        self.loss_parameters.noise_model_likelihood = self.noise_model_likelihood
-        self.loss_parameters.gaussian_likelihood = self.gaussian_likelihood
-        self.loss_parameters.noise_model = self.noise_model
```

Comment:
So let me try to clarify. In the current version of the code, `LVAELossParameters` contains only the bare minimum of parameters that are needed for the loss computation. This means that all the fields that you see removed there are actually removed from the `dataclass` since not needed in any place of the loss computation.
Specifically, these are the "groups" of parameters needed in `LVAELossParameters`:
- likelihood objects
- a set of weights for the different loss terms (to give different importance to different loss terms)
- some additional parameters for KL loss annihilation (not used in ashesh's works, but I would leave it there since I think it is something pretty common to have for VAE-based algoriithms)
- some extra params (e.g., `non_stochastic`, `current_epoch`, ...)

Reply:
hmm ok, I see thanks. Anyway looks good for now! We will take this refactoring process step by step.

Reply:
I'll put a big TODO so that we do not forget about this :)

Reply:
Nice, thanks!